PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -1328,3 +1328,26 @@ int nbDigitsFromNbLines(size_t nbLines)
 	}
 	return nbDigits;
 }
+
+generic_string getDateTimeStrFrom(const generic_string& dateTimeFormat, const SYSTEMTIME& st)

```

Comment:
Why is it needed?
We can just use `GetDateFormatEx` here as well. As for me, it is more powerful and stable than manual string parsing.
You can call it with `LOCALE_NAME_INVARIANT`.
If users like it, we even could expose locale settings, etc in preferences.

Reply:
GetDateFormatEx is only for Date, with the time it's necessary to use GetTimeFormatEx.
For format "hh:mm:ss YYYY-MM-DD", we still need to parse the format string to split it into 2 format.
Here my goal is to provide an easy format - so it's easy to understand for user and easy to implement for dev.